### ISSUE TITLE
[2.0.4] Latest Merge breaks during build/compilation

### ISSUE THREAD

**Issue by virgofx** [phalcon/cphalcon#10598]
Note: This isn't the first time merging has broken the build. Please ensure master stays stable or designate another branch for production users to use as stable. 

Additionally, get a process in place that CI's and tests building/compiling.

CentOS 7
Nginx
PHP5.6

```
/bin/sh /tmp/phalcon/build/64bits/libtool --mode=compile gcc  -I. -I/tmp/phalcon/build/64bits -DPHP_ATOM_INC -I/tmp/phalcon/build/64bits/include -I/tmp/phalcon/build/64bits/main -I/tmp/phalcon/build/64bits -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib  -DPHALCON_RELEASE -DHAVE_CONFIG_H  -march=native -mtune=native -O2 -finline-functions -fomit-frame-pointer -fvisibility=hidden   -c /tmp/phalcon/build/64bits/phalcon.zep.c -o phalcon.lo 
libtool: compile:  gcc -I. -I/tmp/phalcon/build/64bits -DPHP_ATOM_INC -I/tmp/phalcon/build/64bits/include -I/tmp/phalcon/build/64bits/main -I/tmp/phalcon/build/64bits -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib -DPHALCON_RELEASE -DHAVE_CONFIG_H -march=native -mtune=native -O2 -finline-functions -fomit-frame-pointer -fvisibility=hidden -c /tmp/phalcon/build/64bits/phalcon.zep.c  -fPIC -DPIC -o .libs/phalcon.o
/tmp/phalcon/build/64bits/phalcon.zep.c:66405:26: error: redefinition of 'zephir_init_properties'
 static zend_object_value zephir_init_properties(zend_class_entry *class_type TSRMLS_DC) {
                          ^
/tmp/phalcon/build/64bits/phalcon.zep.c:33174:26: note: previous definition of 'zephir_init_properties' was here
 static zend_object_value zephir_init_properties(zend_class_entry *class_type TSRMLS_DC) {
                          ^
/tmp/phalcon/build/64bits/phalcon.zep.c:97273:26: error: redefinition of 'zephir_init_properties'
 static zend_object_value zephir_init_properties(zend_class_entry *class_type TSRMLS_DC) {
                          ^
/tmp/phalcon/build/64bits/phalcon.zep.c:66405:26: note: previous definition of 'zephir_init_properties' was here
 static zend_object_value zephir_init_properties(zend_class_entry *class_type TSRMLS_DC) {
                          ^
/tmp/phalcon/build/64bits/phalcon.zep.c:111339:26: error: redefinition of 'zephir_init_properties'
 static zend_object_value zephir_init_properties(zend_class_entry *class_type TSRMLS_DC) {
                          ^
/tmp/phalcon/build/64bits/phalcon.zep.c:97273:26: note: previous definition of 'zephir_init_properties' was here
 static zend_object_value zephir_init_properties(zend_class_entry *class_type TSRMLS_DC) {
                          ^
/tmp/phalcon/build/64bits/phalcon.zep.c:111467:26: error: redefinition of 'zephir_init_properties'
 static zend_object_value zephir_init_properties(zend_class_entry *class_type TSRMLS_DC) {
                          ^
/tmp/phalcon/build/64bits/phalcon.zep.c:111339:26: note: previous definition of 'zephir_init_properties' was here
 static zend_object_value zephir_init_properties(zend_class_entry *class_type TSRMLS_DC) {
                          ^
/tmp/phalcon/build/64bits/phalcon.zep.c:111615:26: error: redefinition of 'zephir_init_properties'
 static zend_object_value zephir_init_properties(zend_class_entry *class_type TSRMLS_DC) {
                          ^
/tmp/phalcon/build/64bits/phalcon.zep.c:111467:26: note: previous definition of 'zephir_init_properties' was here
 static zend_object_value zephir_init_properties(zend_class_entry *class_type TSRMLS_DC) {
                          ^
/tmp/phalcon/build/64bits/phalcon.zep.c:111757:26: error: redefinition of 'zephir_init_properties'
 static zend_object_value zephir_init_properties(zend_class_entry *class_type TSRMLS_DC) {
                          ^
/tmp/phalcon/build/64bits/phalcon.zep.c:111615:26: note: previous definition of 'zephir_init_properties' was here
 static zend_object_value zephir_init_properties(zend_class_entry *class_type TSRMLS_DC) {
                          ^
/tmp/phalcon/build/64bits/phalcon.zep.c:111864:26: error: redefinition of 'zephir_init_properties'
 static zend_object_value zephir_init_properties(zend_class_entry *class_type TSRMLS_DC) {
                          ^
/tmp/phalcon/build/64bits/phalcon.zep.c:111757:26: note: previous definition of 'zephir_init_properties' was here
 static zend_object_value zephir_init_properties(zend_class_entry *class_type TSRMLS_DC) {
                          ^
/tmp/phalcon/build/64bits/phalcon.zep.c:111967:26: error: redefinition of 'zephir_init_properties'
 static zend_object_value zephir_init_properties(zend_class_entry *class_type TSRMLS_DC) {
                          ^
/tmp/phalcon/build/64bits/phalcon.zep.c:111864:26: note: previous definition of 'zephir_init_properties' was here
 static zend_object_value zephir_init_properties(zend_class_entry *class_type TSRMLS_DC) {
                          ^
/tmp/phalcon/build/64bits/phalcon.zep.c:112099:26: error: redefinition of 'zephir_init_properties'
 static zend_object_value zephir_init_properties(zend_class_entry *class_type TSRMLS_DC) {
                          ^
/tmp/phalcon/build/64bits/phalcon.zep.c:111967:26: note: previous definition of 'zephir_init_properties' was here
 static zend_object_value zephir_init_properties(zend_class_entry *class_type TSRMLS_DC) {
                          ^
/tmp/phalcon/build/64bits/phalcon.zep.c:112270:26: error: redefinition of 'zephir_init_properties'
 static zend_object_value zephir_init_properties(zend_class_entry *class_type TSRMLS_DC) {
                          ^
/tmp/phalcon/build/64bits/phalcon.zep.c:112099:26: note: previous definition of 'zephir_init_properties' was here
 static zend_object_value zephir_init_properties(zend_class_entry *class_type TSRMLS_DC) {
                          ^
/tmp/phalcon/build/64bits/phalcon.zep.c:112550:26: error: redefinition of 'zephir_init_properties'
 static zend_object_value zephir_init_properties(zend_class_entry *class_type TSRMLS_DC) {
                          ^
/tmp/phalcon/build/64bits/phalcon.zep.c:112270:26: note: previous definition of 'zephir_init_properties' was here
 static zend_object_value zephir_init_properties(zend_class_entry *class_type TSRMLS_DC) {
                          ^
/tmp/phalcon/build/64bits/phalcon.zep.c:112661:26: error: redefinition of 'zephir_init_properties'
 static zend_object_value zephir_init_properties(zend_class_entry *class_type TSRMLS_DC) {
                          ^
/tmp/phalcon/build/64bits/phalcon.zep.c:112550:26: note: previous definition of 'zephir_init_properties' was here
 static zend_object_value zephir_init_properties(zend_class_entry *class_type TSRMLS_DC) {
                          ^
/tmp/phalcon/build/64bits/phalcon.zep.c:132561:26: error: redefinition of 'zephir_init_properties'
 static zend_object_value zephir_init_properties(zend_class_entry *class_type TSRMLS_DC) {
                          ^
/tmp/phalcon/build/64bits/phalcon.zep.c:112661:26: note: previous definition of 'zephir_init_properties' was here
 static zend_object_value zephir_init_properties(zend_class_entry *class_type TSRMLS_DC) {
                          ^
make: *** [phalcon.lo] Error 1
```

**Comment by andresgutierrez** [comment]
I'm so sorry, working on this

**Comment by virgofx** [comment]
@andresgutierrez No prob any way we can add a unit test for the build process to ensure everything is passing CI to prevent future issues?

**Comment by virgofx** [comment]
Confirm fixed in latest commit.

**Comment by dschissler** [comment]
The official 2.0.4 download works for me.

**Comment by virgofx** [comment]
I'm able to build successfully and install; however, the phalcon application returns a blank screen. It started happening with the latest build. No errors are thrown anywhere. Just an empty response.

**Comment by dschissler** [comment]
@virgofx I always like to go straight to my index.php and to start adding in error_log calls when that sort of thing happens.  Then I move to the bootstrap file.

**Comment by virgofx** [comment]
@dschissler --- I started doing that:

* Regular php pages work (phpinfo)
* index.php -> Echo successfully
* bootstrap.php --> Successfully all the way up until: `echo $application->handle()->getContent();`
* Tested to see if it was getting into the view -- Nope, never got into `/views/index/index.php`

What's weird, is nothing has changed from my code base other than installing the latest version.

**Comment by dschissler** [comment]
In some instances (I think because of closures) the exception is eaten.  Perhaps add in the method for the very first event in the controller (I'm not sure what that is) and see that logs anything.  Then perhaps try any dispatcher plugins.

**Comment by virgofx** [comment]
Will keep investigating; however, I'm pretty sure something changed in 2.0.4 that has broken something somewhere as my applications (both) are now not working.

**Comment by dschissler** [comment]
@virgofx Are you on Ubuntu 14.04 by chance?

Yesterday I installed a whole bunch of updates that required me to update.  Now I'm having a strange issue on my framework where 2.0.3 is giving me a white screen and 2.0.4 is not accepting the security token.

**Comment by virgofx** [comment]
I've tested on CentOS and Ubuntu -- moving from 2.0.3 to 2.04 I have confirmed that the Application is not displaying the view at all. 

If for example, inside IndexController, I use `$this->view->render('index', 'index');` .... then the view will display; however, this should not have to be required.

Still debugging ...

**Comment by dschissler** [comment]
I've been changing a lot of things lately so its definitely possible that there is a bug in my code.  What is strange though is that the crashes exposes different behavior on 2.0.3 and 2.0.4.

**Comment by dschissler** [comment]
I figured out that 2.0.3 does work correctly for me.  I had forgotten to change `phqLiterals` back to true since it was only completely fixed in 2.0.4.  So just confusion there.

However!  Handling the behavior of security tokens is definitely different in 2.0.4.  Investigating...

**Comment by virgofx** [comment]
Found the issue -- if an EventsManager is set on the application --- the view will not display; otherwise it will.  Something to do with the code in the checkins that impact the application and viewRender.

**Comment by dschissler** [comment]
I changed my security token checking code to get it to work with Phalcon 2.0.4.

**Comment by dschissler** [comment]
@virgofx If its something that can be worked around then perhaps you should make a new issue for your bug.

**Comment by virgofx** [comment]
@dschissler I don't mind making a bug ... but clearly something has been broken in 2.0.4. I'll make a new issue.

**Comment by dschissler** [comment]
New features, new bugs.  But Phalcon 2.0.4 builds now.

**Comment by virgofx** [comment]
Yes ... I'm all in favor of new features ... would just prefer that there is a stable branch that is more thoroughly tested for those that use Phalcon in production. Either a better merge process, more CI, better tests ... just something to ensure stability.

**Comment by virgofx** [comment]
Closing as new build doesn't break anymore.

**Comment by dschissler** [comment]
@virgofx  I'd say that is just sticking with version 1.3 for the moment.  Phalcon 2 is still *dot 0* at the moment.  So many new 2.0 and even older 1.* bugs have been fixed from 2.0.0 to 2.0.4 that I'm very happy to have bits of breakage at this step.

Maybe though 2.1.0 should become a bit more conservative though?
